tests/storage/test_storage.py

```diff
@@ -748,6 +748,28 @@ def test_fetch_url_with_version(self):
         fetch_url = store.fetch_url("some/key", version_id="some-version")
         assert fetch_url == "/media-url/bucket-name/some/some-version/key"
 
+    @mock.patch.object(storage.S3FileStore, "_get_boto_object")
+    def test_get_last_modified_given_path(self, get_boto_object):
+        """Fetch file should act on the given path.
+
+        It should not add the subdirectory to the start of the key.
+        """
+        bucket_name = "some-bucket"
+        directory = "folder"
+        store = storage.S3SubdirectoryFileStore(f"s3://{bucket_name}/{directory}")
+        base_key = "my/file.txt"
+
+        _last_modified = get_boto_object().last_modified
+
+        full_key = store.get_key(base_key).key
+        assert full_key == os.path.join(directory, base_key)
+
+        s3object = storage.S3Object(bucket_name, full_key)
+
+        store.get_last_modified(s3_object=s3object)
+
+        assert store.get_last_modified(s3_object=s3object) == _last_modified
```

Comment:
Can you add this assertion to make sure we return a `datetime.datetime`?
`assert type(store.get_last_modified(s3_object=s3object)) == datetime.datetime`

Reply:
❓ Isn't that exactly what `mypy` is for?

I'm not exactly sure what this test is doing It gets the last modified date from the object being tested, calls a method then checks that it returns what it said?

This feels like it's testing that `mock.patch.object` works not the code under test. I feel like we should be _setting_ something in `last_modifed` and make sure it returns what we expect. The way this is written now it seems like it:
- Asks the boto object what it's going to return for last modified
- Calls the boto object last modified
- Verify it returns what it said it would

What does this actually test?